bitcoin/src/blockdata/transaction.rs

```diff
@@ -1270,8 +1304,8 @@ impl InputWeightPrediction {
         InputWeightPrediction { script_size, witness_size }
     }
 
-    /// Tallies the total weight added to a transaction by an input with this weight prediction,
-    /// not counting potential witness flag bytes or the witness count varint.
+    /// Computes the **signature weight** added to a transaction by an input with this weight prediction,
+    /// not counting the prevout (txid, index), sequence, potential witness flag bytes or the witness count varint.
```

Comment:
It seems like `signature_weight()` would be a better name for this method eventually.

Reply:
Agreed, I suggested deprecating in the top comment but if it'll stay exposed it would be good to change the name to avoid confusion.

Reply:
I would prefer `witness_weight` or something. Unsure. `signature_weight` may suggest to some users that it's specifically about EC signatures.

In any case, this PR looks good as-is.